### PR TITLE
Adds the capillary laying operation tool.

### PR DESCRIPTION
**Pull request by DrSnips**
From #16737
This adds the Capillary Laying Operation Tool (aka CLOT) and a special surgery step to go along with it. The CLOT is intended to be used as an internal bleeding treatment tool for surgery. It acts like a faster fixovein as well as having a secondary function that allows it to inject biofoam to help stall the internal bleeding you're likely going to treat. The biofoam injection comes at a small cost. It eats rolls of gauze or trauma kits in order to make biofoam at a rate of 1 roll to 1u and 1 trauma kit to 5u. It has a 5u capacity limit. On top of the cost it can also only be used during surgery because the way that it administers biofoam is by a surgery step so you can't use it in the field very well. This step can be done at any time during the surgery. In order to prevent an easy method to get biofoam out of the CLOT there technically isn't any reagents in the tool but rather just a value that it converts to reagents during the surgery step.

:cl:
 * rscadd: Added the capillary laying operation tool (CLOT). An advanced version of the fixOvein. Using the CLOT in hand will toggle it to a secondary state where it can inject biofoam during surgery (the person needs to be on a valid surgery table/surface). You need to feed it rolls of gauze to fill it with biofoam at a rate of 1u per roll. Advanced trauma kits will make 5u per kit.

### CODE REVIEW

**Comment by DrSnips** [comment]
The reason for the lack of a protolathe design here is because of #17729. The design is in there and you just remove the commented section.

**Comment by iFlashYou** [comment]
finally

**Comment by iFlashYou** [comment]
Merge when?

File: code/game/objects/items/weapons/surgery_tools.dm [hunk 1]
```diff
@@ -335,6 +336,57 @@
...
+	sharpness_flags = null
+	surgery_speed = 0.5
+	origin_tech = Tc_MATERIALS + "=5;" + Tc_BIOTECH + "=5;" + Tc_ENGINEERING + "=4"
+	var/foam = 0
+
+/obj/item/weapon/FixOVein/clot/examine(mob/user)
+	..()
+	if(foam)
+		to_chat(user, "\The [src] contains [foam] unit[foam > 1 ? "s" : ""] of biofoam.")
+	else
```
Comment (MadmanMartian) [inline code/game/objects/items/weapons/surgery_tools.dm]: The If and everything in the else seems rather redundant

File: code/modules/surgery/generic.dm [hunk 2]
```diff
@@ -441,3 +441,52 @@
...
+	allowed_tools = list(
+	/obj/item/weapon/FixOVein/clot = 100,
+	)
+
+	priority = 0.1 //Tries to inject biofoam before other steps
+
+	min_duration = 10
+	max_duration = 20
+
+/datum/surgery_step/generic/injectfoam/can_use(mob/living/user, mob/living/carbon/human/target, target_zone, obj/item/tool)
```
Comment (MadmanMartian) [inline code/modules/surgery/generic.dm:463]: Need to check if the tool actually has foam to inject

**Review by MadmanMartian — CHANGES_REQUESTED** [review]

File: code/modules/surgery/generic.dm (hunk 2 above)
Comment (DrSnips) [reply code/modules/surgery/generic.dm:463]: It can only do this step if it has sharpness and it olny has sharpness in injection mode.

**Review by MadmanMartian — APPROVED** [review]

**Pull request merged**